File: PHP/8php 2/greet_jenny.php
<?php
/*
|--------------------------------------------------------------------------
| Jenny's secret message
|--------------------------------------------------------------------------
|
*/

function greet($name) {
   return $name == 'Johnny' ? 'Hello, my love!' : "Hello, {$name}!";
}


function greet($name) {
   if ($name == 'Johnny') {
      return 'Hello, my love!';
   } else {
      return "Hello, $name!";    
   }
}


function greet(string $name): string {
   return sprintf('Hello, %s!', $name === 'Johnny' ? 'my love' : $name);
}


print_r(greet("Jim"). " Hello, Jim!
");
print_r(greet("Jane"). " Hello, Jane!
");
print_r(greet("Simon"). " Hello, Simon!
");
print_r(greet("Johnny"). " Hello, my love!
");

File: Python/8py 1/greet_jenny.py
"""
|--------------------------------------------------------------------------
| Jenny's secret message
|--------------------------------------------------------------------------
|
"""

def greet(name):
   return 'Hello, my love!' if name == 'Johnny' else f'Hello, {name}!'


greet = lambda n: "Hello, {}!".format(n.replace("Johnny","my love"))


def greet(name):
   if name != 'Johnny':
      return "Hello, {name}!".format(name=name)
   else:
      return "Hello, my love!"


def greet(name):
   return "Hello, {name}!".format(name = ('my love' if name == 'Johnny' else name))


def greet(name):
   return "Hello, {name}!".format(name=name.replace("Johnny", "my love"))    
 

print(greet("James"), "Hello, James!")
print(greet("Jane"), "Hello, Jane!")
print(greet("Jim"), "Hello, Jim!")
print(greet("Johnny"), "Hello, my love!")